build.go

```diff
@@ -82,6 +110,41 @@ func Build(
 			return packit.BuildResult{}, fmt.Errorf("could not remove temp directory: %w", err)
 		}
 
+		if globalNugetPath != "" {
+			err = symlinker.Unlink(filepath.Join(homeDir, ".nuget", "NuGet", "NuGet.Config"))
+			if err != nil {
+				return packit.BuildResult{}, err
+			}
+		}
+
```

Comment:
Why was unlink functionality written when it appears that remove the link is sufficient? Is there an edge case that y'all were worried about? Feels like a lot of additional logic for something that we highly control.

Reply:
I'm not quite sure what you mean by "that remove the link is sufficient".

When I implemented the Link/Unlink interface in the npm install and yarn install buildpacks, my intent was to abstract away the whole symlinking process from the build function, so we wouldn't have to test filesystem operations related to the symlinks within `build_test.go`. Does that answer your question?

Reply:
That's fair enough. I just feels a little excessive by something that could be done with just `os.Remove()` functionality. Is there a reason that validating that the path given to the function is a link is necessary though? If this is purely to abstract away symlinking for testing purpose I think that is fine I am just trying to get at the intention.

Reply:
Are you asking why we also have to check `globalNugetPath` before doing this? Not sure I understand the last comment

Reply:
I think in this case we could get rid of 
```
if fileInfo.Mode()&os.ModeSymlink != os.ModeSymlink {
		return fmt.Errorf("cannot unlink %s because it is not a symlink", path)
	}
```
in the `Unlink` function, since we have the `if globalNugetPath != ""` check here.

Reply:
That's fine by me. I'm not too worried about it. When I wrote the validation, my intent was to make sure `Unlink()` did exactly what its name implied and not more.

Reply:
I think that you could keep `Unlink()` the way it is I was just curious why the logic was there. If this is preferable and easier to test I have very little reason not to let it through.